### PR TITLE
Make systemd unit start detect already running unit

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

During the plan phase, see if the unit is running, if it is, skip doing anything. Part of #35

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

File: src/action/linux/start_systemd_unit.rs [hunk 1]
```diff
@@ -17,11 +17,27 @@ pub struct StartSystemdUnit {
...
+        // If `systemctl is-active $UNIT` has exit status 3, it doesn't exist
+        let output = Command::new("systemctl")
+            .arg("is-active")
+            .arg(unit)
+            .output()
+            .await
+            .map_err(ActionError::Command)?;
+
+        let state = if output.status.success() {
+            tracing::debug!("Starting systemd unit `{}` already complete", unit,);
```
Comment (cole-h) [inline src/action/linux/start_systemd_unit.rs:30]: Nit:

```suggestion
            tracing::debug!("Starting systemd unit `{}` already complete", unit);
```

File: src/action/linux/start_systemd_unit.rs [hunk 2]
```diff
@@ -17,11 +17,27 @@ pub struct StartSystemdUnit {
 impl StartSystemdUnit {
     #[tracing::instrument(level = "debug", skip_all)]
     pub async fn plan(unit: impl AsRef<str>) -> Result<StatefulAction<Self>, ActionError> {
+        let unit = unit.as_ref();
+        // If `systemctl is-active $UNIT` has exit status 3, it doesn't exist
```
Comment (cole-h) [inline src/action/linux/start_systemd_unit.rs:21]: Does this comment still need to be here? It doesn't look like we worry about "exit status 3" anywhere, so I don't think we need to worry about documenting what it means.

```suggestion
```

File: src/action/linux/start_systemd_unit.rs (hunk 1 above)
Comment (Hoverbear) [reply src/action/linux/start_systemd_unit.rs:30]: How does Rustfmt not get this...

Comment (cole-h) [reply src/action/linux/start_systemd_unit.rs:30]: Macros are hard for rustfmt, unfortunately.

**Review by cole-h — APPROVED** [review]

**Comment by Hoverbear** [comment]
I'm just gonna merge this since our API rate limiting bug is once again present and this doesn't touch Mac.

**Pull request merged**